public/video-ui/src/components/VideoUpload/VideoUpload.js

```diff
@@ -63,30 +63,45 @@ class VideoUpload extends React.Component {
     }
   };
 
-  startUpload = () => {
+  startUpload = (selfHost) => {
     if(this.props.video && this.state.file) {
       const atomId = this.props.video.id;
 
       this.props.uploadActions.startUpload(atomId, this.state.file, () => {
         // on complete
         this.props.uploadActions.getUploads(atomId);
-      });
+      }, selfHost);
     }
   };
 
+
+
   renderButtons(uploading) {
+    const addAssetPermission = getStore().getState().config.permissions.addAsset
+    const addSelfHostedAssetPermission = getStore().getState().config.permissions.addSelfHostedAsset
     if (uploading) {
       return false;
-    } else {
-      return <button type="button" className="btn button__secondary__assets" disabled={!this.state.file} onClick={this.startUpload}>
-        <Icon icon="backup">Upload</Icon>
-      </button>;
     }
+    else if (addAssetPermission && addSelfHostedAssetPermission) {
+      return <div> {this.renderStartUpload(false, 'Upload')} {this.renderStartUpload(true, 'Upload avoiding YouTube')} </div>;
+    }
+    else if (addAssetPermission) {
+      return this.renderStartUpload(false, 'Upload');
+    }
+    else if (addSelfHostedAssetPermission) {
+      return this.renderStartUpload(true, 'Upload avoiding YouTube');
+    }
```

Comment:
Could you write separate functions for rendering buttons for adding youtube assets and self hosted assets that return false if correct permissions do not exist or call `this.renderStartUpload` with the correct arguments if they do exist? Then you could just do something like
`<div>{this.renderAddAsset()} {this.renderAddHostedAsset()}</div>`